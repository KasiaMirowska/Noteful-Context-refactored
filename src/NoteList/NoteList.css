.note-list {
    margin: 15px;
}
.noteList-header-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
} 
.li2 {
    color: black;
    display: flex;
    padding: 10px;
}
.link2 {
    text-decoration: none;
    color:rgb(150, 74, 3);
    margin: 0 auto;
  }
.noteList-header {
    font-family: 'Barriecito', cursive;
    color:rgb(204, 80, 8);
    padding: 10px;
    font-size: 2em;
  
} 
@media (min-width: 600px) {
    .note-list {
        flex: 2;
        margin: 0;
    }
  }  
 