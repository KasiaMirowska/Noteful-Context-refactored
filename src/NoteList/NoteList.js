import React from 'react';
import NotefulContext from '../NotefulContext';
import './NoteList.css';
import NoteListItem from '../NoteListItem/NoteListItem';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class NoteList extends React.Component {
  static contextType = NotefulContext;

  filterNotesByFolder = () => {
    const notes = this.context.notes
    if (!this.props.match.params.folderId) {
      return notes;
    }

    return notes.filter((note) => {
      if (note.folder === Number(this.props.match.params.folderId)) {
        return note;
      }
    })
  }

  render() {
    const notes = this.filterNotesByFolder().map(note => {
      return <NoteListItem
        key={note.id}
        id={note.id}
        name={note.name}
        modified={note.date_created}
        folderId={note.folder}
        content={note.content}
      />
    })

    return (
      <div className='note-list'>
        <div className='noteList-header-container'>
        <h2 className='noteList-header'>Notes</h2>
        <Link to='/newNote'>
          <button className='add-note'>Add Note</button>
        </Link>
        </div>

        {notes}

      </div>
    )
  }

}
NoteList.propTypes = {
  history: PropTypes.objectOf(PropTypes.shape({
    length: PropTypes.number,
    action: PropTypes.string,
    location: PropTypes.object,
    createHref: PropTypes.object,
    push: PropTypes.object,
  })),
  location: PropTypes.objectOf(PropTypes.shape({
    pathname: PropTypes.string,
    search: PropTypes.string,
    hash: PropTypes.string,
    key: PropTypes.string,
  })),
  match: PropTypes.objectOf(PropTypes.shape({
    path: PropTypes.string,
    url: PropTypes.string,
    isExact: PropTypes.boolean,
    params: PropTypes.objectOf(PropTypes.shape({
      path: PropTypes.string,
      url: PropTypes.string,
    }))
  })),
}