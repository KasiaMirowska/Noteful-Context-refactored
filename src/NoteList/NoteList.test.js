import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NoteList from './NoteList';
import { BrowserRouter } from 'react-router-dom';

describe('NoteList component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><NoteList match={{params:{folderId : 1}}}/></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        expect(renderer.create(<BrowserRouter><NoteList match={{params:{folderId : 1}}}/></BrowserRouter>).toJSON()).toMatchSnapshot();
    });
});



  