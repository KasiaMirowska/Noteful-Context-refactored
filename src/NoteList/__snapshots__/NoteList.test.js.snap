// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList component renders the UI as expected 1`] = `
<div
  className="note-list"
>
  <div
    className="noteList-header-container"
  >
    <h2
      className="noteList-header"
    >
      Notes
    </h2>
    <a
      href="/newNote"
      onClick={[Function]}
    >
      <button
        className="add-note"
      >
        Add Note
      </button>
    </a>
  </div>
</div>
`;
